apps/web/src/presentation/react/components/NetworkSwitch.tsx

```diff
@@ -0,0 +1,111 @@
+import { ComponentProps, MutableRefObject, ReactNode, useState } from 'react'
+import { Menu } from './Menu'
+import { useOnClickOutside } from '../hooks/useOnClickOutside'
+import { CheckMarkIcon } from './icons/CheckMark'
+import { Chevron } from './icons/Chevron'
+import React from 'react'
+
+type Props = {
+  className?: string
+  networks: string[]
+  selectedNetwork: string
+  selectNetwork: (value: string) => void
+}
+
+const MenuContainer = ({
+  children,
+  refProp,
+  className,
+  ...props
+}: { children: ReactNode } & {
+  refProp: React.RefObject<HTMLDivElement> | MutableRefObject<HTMLDivElement>;
+} & ComponentProps<'div'>) => (
+  <div
+    {...props}
+    className={`cursor-pointer rounded-lg p-2 h-10 bg-neutral-100 flex items-center justify-center ${className ?? ''}`}
+    ref={refProp}
+  >
+    {children}
+  </div>
+);
+
+const ItemText = ({
+  selected = false,
+  text = ''}) => (
+  <span
+    className={`text-base font-medium capitalize transition-colors duration-300
+      group-hover/item:text-[#1A1B23] ${
+        selected ? 'text-neutral-950' : 'text-[#1A1B23]'
+      }`}
+  >
+    {text}
+  </span>
+)
+
+const Row = (props: { children: ReactNode } & ComponentProps<'div'>) => (
+  <div className="flex items-center gap-x-2" {...props} />
+)
```

Comment:
_I think_  ` ComponentProps<'div'>` already includes `children`

```suggestion
const Row = (props: ComponentProps<'div'>) => (
  <div className="flex items-center gap-x-2" {...props} />
)
```

Reply:
Indeed it includes. Thanks!